Kitodo/src/main/java/org/kitodo/production/KitodoProduction.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.jar.Manifest;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.production.helper.tasks.TaskManager;
+import org.kitodo.production.interfaces.activemq.ActiveMQDirector;
+import org.kitodo.production.security.SecurityUserDetails;
+import org.kitodo.production.services.ServiceManager;
+import org.springframework.security.core.context.SecurityContextImpl;
+
+/**
+ * Kitodo.Production is the workflow management module of the Kitodo suite. It
+ * supports the digitization process of various types of materials.
+ */
+@WebListener
+public class KitodoProduction implements ServletContextListener, HttpSessionListener {
+    private static final Logger logger = LogManager.getLogger(KitodoProduction.class);
+    private static CompletableFuture<KitodoProduction> instance = new CompletableFuture<>();
+    private ServletContext context;
+    private Optional<Manifest> manifest;
+    private KitodoVersion version = new KitodoVersion();
+    private ActiveMQDirector activeMQDirector;
+
+    /* This is the main() function. The class is instantiated by the servlet
+     * container, which then calls this function. */
+    @Override
+    public void contextInitialized(ServletContextEvent event) {
+        context = event.getServletContext();
+        manifest = retrieveManifestFileAsStream(context);
+        manifest.ifPresent(version::setupFromManifest);
+        instance.complete(this);
+        startActiveMQ();
+    }
+
+    private static final Optional<Manifest> retrieveManifestFileAsStream(ServletContext context) {
```

Comment:
I once did read that one should make all methods used in the constructor final, because if they are overloaded in a subclass, it could have confusing effects. I always do that; if the methods are under the constructor, you can see nicely in the “outline” what belongs to the constructors and where the actual class begins. For me, that is an improvement in the clarity of the code. In this case, it is particularly exciting because the “constructor” is not a constructor in the Java sense, but a method that is called by the Servlet Container to construct the class.

![outline_](https://github.com/user-attachments/assets/8dd00119-6849-4e2c-9de7-d25d495debef)

Reply:
I am not convinced this having a structure outline in your IDE is a good enough reason to overrule this formal recommendation, but I can live with it if it really helps you with.